eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/runner/EvaAccessionJobLauncherCommandLineRunner.java

```diff
@@ -0,0 +1,160 @@
+ */
+@Component
+public class EvaAccessionJobLauncherCommandLineRunner extends JobLauncherCommandLineRunner implements
+        ApplicationEventPublisherAware, ExitCodeGenerator {
+
+    private static final Logger logger = LoggerFactory.getLogger(EvaAccessionJobLauncherCommandLineRunner.class);
+
+    public static final String SPRING_BATCH_JOB_NAME_PROPERTY = "spring.batch.job.names";
+
+    public static final int EXIT_WITHOUT_ERRORS = 0;
+
+    public static final int EXIT_WITH_ERRORS = 1;
+
+    @Value("${" + SPRING_BATCH_JOB_NAME_PROPERTY + ":#{null}}")
+    private String jobName;
+
+    private Collection<Job> jobs;
+
+    private JobRepository jobRepository;
+
+    @Autowired
+    private JobExecutionApplicationListener jobExecutionApplicationListener;
+
+    @Autowired
+    private InputParameters inputParameters;
+
+    private boolean abnormalExit;
+
+    public EvaAccessionJobLauncherCommandLineRunner(JobLauncher jobLauncher, JobExplorer jobExplorer,
+                                                    JobRepository jobRepository) {
+        super(jobLauncher, jobExplorer);
+        this.jobRepository = jobRepository;
+    }
+
+
+    @Autowired(required = false)
+    public void setJobs(Collection<Job> jobs) {
+        this.jobs = jobs;
+    }
+
+    @Override
+    public void setJobNames(String jobName) {
+        this.jobName = jobName;
+        super.setJobNames(jobName);
+    }
+
+    @Override
+    public int getExitCode() {
+        if (!abnormalExit && jobExecutionApplicationListener.isJobExecutionComplete()) {
+            return EXIT_WITHOUT_ERRORS;
+        } else {
+            return EXIT_WITH_ERRORS;
+        }
+    }
+
+    @Override
+    public void run(String... args) throws JobExecutionException {
+        try {
+            abnormalExit = false;
+
+            // TODO: different jobs can have different parameters
+            JobParameters jobParameters = inputParameters.toJobParameters();
+
+            JobStatusManager.checkIfJobNameHasBeenDefined(jobName);
+            JobStatusManager.checkIfPropertiesHaveBeenProvided(jobParameters);
+            if (inputParameters.isForceRestart()) {
+                markPreviousJobAsFailed(jobParameters);
+            }
+            launchJob(jobParameters);
+        } catch (NoJobToExecuteException | NoParametersHaveBeenPassedException | NoPreviousJobExecutionException
+                | UnknownJobException | JobParametersInvalidException | JobExecutionAlreadyRunningException e) {
+            logger.error(e.getMessage());
+            logger.debug("Error trace", e);
+            abnormalExit = true;
+        }
+
+    }
+
+    private void launchJob(JobParameters jobParameters) throws JobExecutionException, UnknownJobException {
+        for (Job job : this.jobs) {
+            // TODO is PatternMatchUtils necessary because the jobs discovered by Spring include package name? if not, remove it
```

Comment:
I debugged the tests and the job name didn't include the package, but I think that it can be different when Spring scan for Jobs, so I will try now with the real job